addons/base_action_rule/models/base_action_rule.py

```diff
@@ -0,0 +1,386 @@
+            self.trg_date_id = self.trg_date_range = self.trg_date_range_type = False
+        elif self.kind == 'on_time':
+            self.filter_pre_id = False
+
+    @api.onchange('filter_pre_id')
+    def onchange_filter_pre_id(self):
+        self.filter_pre_domain = self.filter_pre_id.domain
+
+    @api.onchange('filter_id')
+    def onchange_filter_id(self):
+        self.filter_domain = self.filter_id.domain
+
+    @api.model
+    def create(self, vals):
+        base_action_rule = super(BaseActionRule, self).create(vals)
+        self._update_cron()
+        self._update_registry()
+        return base_action_rule
+
+    @api.multi
+    def write(self, vals):
+        res = super(BaseActionRule, self).write(vals)
+        if set(vals) & set(self.CRITICAL_FIELDS):
+            self._update_cron()
+            self._update_registry()
+        return res
+
+    @api.multi
+    def unlink(self):
+        res = super(BaseActionRule, self).unlink()
+        self._update_cron()
+        self._update_registry()
+        return res
+
+    @api.multi
+    def _filter(self, action_filter, record_ids, domain=False):
+        """ Filter the list record_ids that satisfy the domain or the action filter. """
+        self.ensure_one()
+        if record_ids and (domain or action_filter):
+            ctx = dict(self.env.context or {})
+            eval_context = self._get_eval_context()
+            if domain:
+                new_domain = [('id', 'in', record_ids)] + eval(domain, eval_context)
+            elif action_filter:
+                assert self.model == action_filter.model_id, "Filter model different from action rule model"
+                new_domain = [('id', 'in', record_ids)] + eval(action_filter.domain, eval_context)
+                ctx.update(eval(action_filter.context))
+            record_ids = self.env[self.model].with_context(ctx).search(new_domain).ids
+        return record_ids
+
+    @api.multi
+    def _process(self, record_ids):
+        """ process the given action on the records """
+        self.ensure_one()
+        model_records = self.env[self.model].browse(record_ids)
+        # modify records
+        values = {}
+        if 'date_action_last' in model_records._fields:
+            values['date_action_last'] = fields.Datetime.now()
+        if self.act_user_id and 'user_id' in model_records._fields:
+            values['user_id'] = self.act_user_id.id
+        if values:
+            model_records.write(values)
+
+        if self.act_followers and hasattr(model_records, 'message_subscribe'):
+            model_records.message_subscribe(self.act_followers.ids)
+
+        # execute server actions
+        if self.server_action_ids:
+            for record in model_records:
+                self.server_action_ids.with_context(active_model=model_records._name, active_ids=[record.id], active_id=record.id).run()
+        return True
+
+    @api.v8
+    def _register_hook(self):
```

Comment:
If i am wrong then please correct me, @msh-odoo some times this method use from web as well portal side so when combo of dual side and diversion of ```new api``` and ```old api``` use it so its sensible. when all modules of the odoo migrated then from one code migration script all v7 decorated method gonna removed that is why also this code is sensible.